src/main/webapp/app/entities/programming-exercise/programming-exercise-instruction.component.ts

```diff
@@ -0,0 +1,466 @@
+
+    /**
+     * Reset and then render the markdown of the instruction file.
+     */
+    updateMarkdown() {
+        this.steps = [];
+        this.renderedMarkdown = this.markdown.render(this.exercise.problemStatement);
+        // For whatever reason, we have to wait a tick here. The markdown parser should be synchronous...
+        setTimeout(() => this.setUpClickListeners(), 100);
+        this.isLoading = false;
+    }
+
+    /**
+     * Retrieve latest result for the participation/exercise/course combination.
+     */
+    loadLatestResult(): Promise<void> {
+        return new Promise((resolve, reject) => {
+            this.resultService.findResultsForParticipation(this.exercise.course.id, this.exercise.id, this.participation.id).subscribe(
+                (latestResult: any) => {
+                    this.latestResult = latestResult.body.length && latestResult.body[0];
+                    resolve();
+                },
+                err => {
+                    console.log('Error while loading latest results!', err);
+                    reject();
+                },
+            );
+        });
+    }
+
+    /**
+     * @function loadResultDetails
+     * @desc Fetches details for the result (if we received one) => Input latestResult
+     */
+    loadResultsDetails(): Promise<void> {
+        return new Promise((resolve, reject) =>
+            this.resultService.getFeedbackDetailsForResult(this.latestResult.id).subscribe(
+                resultDetails => {
+                    this.resultDetails = resultDetails.body;
+                    this.latestResult.feedbacks = this.resultDetails;
+                    resolve();
+                },
+                err => {
+                    console.log('Error while loading result details!', err);
+                    reject();
+                },
+            ),
+        );
+    }
+
+    /**
+     * @function loadInstructions
+     * @desc Loads the instructions for the programming exercise.
+     * We added the problemStatement later, historically the instructions where a file in the student's repository
+     * This is why we now prefer the problemStatement and if it doesn't exist try to load the readme.
+     */
+    loadInstructions(): Promise<void> {
+        return new Promise((resolve, reject) => {
+            // Historical fallback: Older exercises have an instruction file in the git repo
+            if (this.exercise.problemStatement === undefined) {
+                const participationId = this.showTemplatePartipation ? (this.exercise as ProgrammingExercise).templateParticipation.id : this.participation.id;
+                this.repositoryFileService.get(participationId, 'README.md').subscribe(
+                    fileObj => {
+                        // Old readme files contain unescaped unicode, convert it to make it persistable by the database
+                        this.exercise.problemStatement = fileObj.fileContent.replace(new RegExp(/✅/, 'g'), '&#x2705;');
```

Comment:
This is the fallback for trying to load instructions from the git repo if there is no problemStatement.
To avoid the unicode issue (see above), I replace the checkmark with the escaped code.